titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/Job.java

```diff
@@ -45,11 +46,15 @@
 
     private final List<JobStatus> statusHistory;
 
-    public Job(String id, JobDescriptor<E> jobDescriptor, JobStatus status, List<JobStatus> statusHistory) {
+    private final Version version;
+
+    public Job(String id, JobDescriptor<E> jobDescriptor, JobStatus status, List<JobStatus> statusHistory, Version version) {
         this.id = id;
         this.jobDescriptor = jobDescriptor;
         this.status = status;
         this.statusHistory = CollectionsExt.nullableImmutableCopyOf(statusHistory);
+        // TODO Remove me. Version is a newly introduced field. If not set explicitly, assign a default value
```

Comment:
Just to clarify, the "remove me" is for the handling of null values not the version field itself?

Reply:
Yes, so we do not accidentally create an object that has null Version value.